packages/rx_bloc_cli/example/docs/auth_matrix.md

```diff
@@ -0,0 +1,257 @@
+Then, extend the AuthMatrixPayloadRequest class to create a payload class for the new method, 
+incorporating necessary payload fields and implementing methods for JSON serialization and equatability.
+
+#### Step 1: Add a new case in the AuthMatrixAction enum
+New case in the AuthMatrixAction needs to be added for each new authentication method.
+```dart
+enum AuthMatrixAction {
+  // the other actions...
+  changePassword,
+}
+```
+
+#### Step 2: Add a new case in the AuthMatrixActionExtension
+The payload for the new authentication method needs to be implemented by extending the AuthMatrixPayloadRequest class.
+```dart
+@JsonSerializable()
+class AuthMatrixChangePasswordPayload extends AuthMatrixPayloadRequest
+        with EquatableMixin {
+  AuthMatrixChangePasswordPayload({
+    required this.password,
+  });
+
+  final String password;
+
+  @override
+  String get type => AuthMatrixAction.changePassword.name;
+
+  @override
+  List<Object?> get props => [password];
+
+  factory AuthMatrixChangePasswordPayload.fromJson(Map<String, dynamic> json) =>
+          _$AuthMatrixChangePasswordPayloadFromJson(json);
+
+  @override
+  Map<String, dynamic> payloadToJson() =>
+          _$AuthMatrixChangePasswordPayloadToJson(this);
+}
+```
+
+#### Step 3: Initiate the new authentication action
+```dart
+_service.initiateAuthMatrix(payload: AuthMatrixChangePasswordPayload(password: 'password'));
+```
+# Authentication method
+
+To add a new method to the authentication matrix, such as email verification, first, add a new case `emailVerification` to the `AuthMatrixMethod` enum.
+Create the `AuthMatrixEmailVerificationPayload` class extending `AuthMatrixPayloadRequest` and update the `AuthMatrixMethodExtension` class to parse this new payload.
+Define the `AuthMatrixEmailVerificationRoute` class extending `GoRouteData` to handle routing.
+Finally, add the new route path in the `RoutesPath` class and update the `createAuthMatrixMethodRoute` method to include the new enum case.
+
+
+#### Step 1: Payload - Add a new case in the AuthMatrixMethod enum
+```dart
+enum AuthMatrixMethod {
+  ...
+  @JsonValue('EmailVerification')
+  emailVerification,
+```
+
+#### Step 2: Payload -  Create a class that extends AuthMatrixPayloadRequest and uses the new AuthMatrixMethod
+```dart
+class AuthMatrixEmailVerificationPayload extends AuthMatrixPayloadRequest
+    with EquatableMixin {
+  AuthMatrixEmailVerificationPayload();
+
+  @override
+  String get type => AuthMatrixMethod.emailVerification.name;
+
+  factory AuthMatrixEmailVerificationPayload.fromJson(Map<String, dynamic> json) => AuthMatrixEmailVerificationPayload();
+
+  @override
+  Map<String, dynamic> payloadToJson() => {};
+
+  @override
+  List<Object?> get props => [type];
+
+  @override
+  bool? get stringify => true;
+}
+```
+
+#### Step 3: Payload - Add a new case in the AuthMatrixMethodExtension class to parse the new request method payload 
```

Comment:
The [AuthMatrixMethodExtensions](packages/rx_bloc_cli/mason_templates/bricks/lib_auth_matrix/__brick__/lib/lib_auth_matrix/extensions/auth_matrix_method_extensions.dart) mentioned don't feature the code below

Reply:
The documentation has been updated